test/validators/format.js

```diff
@@ -0,0 +1,106 @@
+const assert = require("assert")
+const { validate, errorCodes } = require("../../src/suma")
+const err = errorCodes
+
+describe("format validation", () => {
+    it('does allow empty values', () => {
+
+        //zipcode regex
+        var pattern = /^[0-9]{8}$/
+        
+        const samples = [
+            {},
+            [],
+            '',
+            ' ',
+            null,
+            undefined
+        ]
+        for (const value of samples) {
+            // given
+            const validations = { format: pattern }
+            // when
+            const ret = validate(value, validations)
+            // then
+            assert.deepStrictEqual(ret, { value: value, errors: [] })
+        }
+    })
+
+    it("allows values that matches the pattern", function() {
+
+       //zipcode regex
+       var pattern = /^[0-9]{8}$/
+        
+       const samples = [
+          "05541030"
+       ]
+
+       for (const value of samples) {
+           // given
+           const validations = { format: pattern }
+           // when
+           const ret = validate(value, validations)
+           // then
+           assert.deepStrictEqual(ret, { value: value, errors: [] })
+       }
+
+      })
+
+
+    it("allows work with multiple validations", function() {
+
+        //zipcode regex
+        var pattern = /^[0-9]{8}$/
+         
+        const samples = [
```

Comment:
is this case, since you have just one value to be tested, I don't think you need variables specially `samples`, since this implies multiple values